corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js

```diff
@@ -539,13 +539,54 @@ hqDefine("cloudcare/js/formplayer/menus/views", function () {
                 L.mapbox.accessToken = token;
-                const mapbox = L.mapbox;
-                const geocoder = mapbox.geocoder('mapbox.places');
 
-                const latLons = this.options.collection.models
-                    .map(model => this.addAddressPin(geocoder, addressMap, this.options.headers, model, addressIndex))
-                    .filter(latLon => latLon);
-                addressMap.fitBounds(latLons, {maxZoom: 8});
+                const latLons = []
+                const markers = []
+                this.options.collection.models
+                    .forEach(model => {
+                        const coordinates = model.attributes.data[addressIndex];
+                        if (coordinates) {
+                            let latLng = coordinates.split(" ").slice(0,2);
+                            if (latLng.length > 1) {
+                                const rowId = `row-${model.id}`;
+                                const marker = L.marker(latLng, {icon: locationIcon});
+                                markers.push(marker);
+                                marker
+                                    .addTo(addressMap)
+                                    .on('click', () => {
+                                        // tiles
+                                        $(`.list-cell-wrapper-style[id!='${rowId}']`)
+                                            .removeClass("highlighted-case");
+                                        // rows
+                                        $(`.case-row[id!='${rowId}']`)
+                                            .removeClass("highlighted-case");
+                                        $(`#${rowId}`)
+                                            .addClass("highlighted-case");
+                                        markers.forEach(m => m.setIcon(locationIcon));
+                                        marker.setIcon(selectedLocationIcon);
+
+                                        $([document.documentElement, document.body]).animate({
+                                            scrollTop: $(`#${rowId}`).offset().top
+                                        }, 500);
+                                    });
+                                latLons.push(latLng);
+                            }
+                        }
+                    });
+
+                navigator.geolocation.getCurrentPosition(
+                    success => {
+                        const coords = success.coords;
+                        const homeLatLng = [coords.latitude, coords.longitude];
+
+                        L.marker(homeLatLng, { icon: homeLocationIcon })
+                            .addTo(addressMap);
+                        latLons.push(homeLatLng);
+                        addressMap.fitBounds(latLons, {maxZoom: 8});
+                    }, () => {
+                        addressMap.fitBounds(latLons, {maxZoom: 8});
```

Comment:
Why is `fitBounds` called twice?

Reply:
oh, is this what happens if `getCurrentPosition` fails?

Reply:
yes